### DISCUSSION TITLE
access ontroly by code behind

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#11328] [Q&A] [answered]
Hello,

my view has a DataGrid with property x:Name set

```
            <DataGrid x:Name="datagrid"
```

It seems that datagrid is always null.

I would expect that datagrid is not null at least after the call of InitializeComponent(). But that's not the case.

I am loading some data into my DataGrid and after I want to bring the last entry into view by

```
        SelectedItem = Data.LastOrDefault();
        if (SelectedItem != null)
        {
            logger.Log(LogLevel.Information, $"dataGrid is null? {View.datagrid==null}");
            View.datagrid?.ScrollIntoView(SelectedItem, null);
        }
```

but unfortunately datagrid is null :-(

```
2023-05-11T02:37:04.815 :info: JHXManager.ViewModels.RechnungenViewModel[0] dataGrid is null? True
```

Bug or Feature?

**Comment by timunie** [comment, marked as answer]
If you are using v0.10.x and you have not the XAMLNamesGenerator installed, you need to find the control by name before accessing it. It will not be auto generated. 

See: https://github.com/avaloniaui/Avalonia.NameGenerator/
https://docs.avaloniaui.net/guides/basics/code-behind#locating-controls

**Reply by drei97** [reply]
FindControl<> is working

But:

XamlNameReferenceGenerator is installed from the beginning. A XamlNamesGenerator is not found, same for Avalonia.NameGenerator. I am not sure if they are all meant to be the same.


![image](https://github.com/AvaloniaUI/Avalonia/assets/68755045/92f47c8e-b566-44b5-a466-5ae837fa8257)


Even if XamlNameReferenceGenerator is installed it seems not to be working by default.

Or do I have to switch it on in a way?

**Reply by timunie** [reply]
It works better in 11.0.0. I didn't use it in 0.10.x (used to use FindControl), so I can't remember how to get it working in 0.10.x

**Reply by drei97** [reply]
OK, since I have only one control to set, FindControl is a suitable workaround.

thnx
